marblecutter/web.py

```diff
@@ -172,14 +175,131 @@ def render_terrarium(z, x, y, scale=1):  # noqa
 
     headers, data = tiling.render_tile(
         tile,
-        POSTGIS_ADAPTER,
+        POSTGIS_CATALOG,
         format=PNG_FORMAT,
         transformation=TERRARIUM_TRANSFORMATION,
         scale=scale)
 
     return data, 200, headers
 
 
+S3_BUCKET = "oin-hotosm"
+
+
+@lru_cache()
+def make_catalog(scene_id, scene_idx, image_id=None):
+    if image_id:
+        return OINMetaCatalog("https://{}.s3.amazonaws.com/{}/{}/{}_meta.json".
+                              format(S3_BUCKET, scene_id, scene_idx, image_id))
+
+    return OAMSceneCatalog("https://{}.s3.amazonaws.com/{}/{}/scene.json".
+                           format(S3_BUCKET, scene_id, scene_idx))
+
+
+@app.route('/<id>/<int:scene_idx>/')
+@app.route('/<id>/<int:scene_idx>/<image_id>/')
+def meta_oam(id, scene_idx, image_id=None):
+    catalog = make_catalog(id, scene_idx, image_id)
+
+    meta = {
+        "bounds": catalog.bounds,
+        "center": catalog.center,
+        "maxzoom": catalog.maxzoom,
+        "minzoom": catalog.minzoom,
+        "name": catalog.name,
+        "tilejson": "2.1.0",
+    }
+
+    with app.app_context():
+        meta["tiles"] = [
+            "{}{{z}}/{{x}}/{{y}}.png".format(
+                url_for(
+                    "meta_oam",
+                    id=id,
+                    scene_idx=scene_idx,
+                    image_id=image_id,
+                    _external=True))
+        ]
+
+    return jsonify(meta)
+
+
+@app.route('/<id>/<int:scene_idx>/wmts')
+@app.route('/<id>/<int:scene_idx>/<image_id>/wmts')
+def wmts(id, scene_idx, image_id=None):
+    catalog = make_catalog(id, scene_idx, image_id)
+
+    provider = "OpenAerialMap"
+    provider_url = "https://openaerialmap.org/"
+
+    if catalog.provider:
+        provider = "{} ({})".format(provider, catalog.provider)
+
+    with app.app_context():
+        base_url = url_for(
+            "meta_oam",
+            id=id,
+            scene_idx=scene_idx,
+            image_id=image_id,
+            _external=True)
+
+        return render_template(
+            'wmts.xml',
+            base_url=base_url,
+            bounds=catalog.bounds,
+            content_type="image/png",
+            ext="png",
+            id=catalog.id,
+            maxzoom=catalog.maxzoom,
+            metadata_url=catalog.metadata_url,
+            minzoom=catalog.minzoom,
+            provider=provider,
+            provider_url=provider_url,
+            title=catalog.name,
+            ), 200, {
+            'Content-Type': 'application/xml'
+        }
+
+
+@app.route('/<id>/<int:scene_idx>/preview')
+@app.route('/<id>/<int:scene_idx>/<image_id>/preview')
+def preview_oam(id, scene_idx, image_id=None):
+    # load the catalog so it will fail if the source doesn't exist
+    make_catalog(id, scene_idx, image_id)
+
+    with app.app_context():
+        return render_template(
+            "preview.html",
```

Comment:
That was an old copy sticking around. The `preview.html` referred to here is in `marblecutter/templates`.